lib/Button.js

```diff
@@ -147,99 +168,107 @@ export default class Button extends Component {
         const textStyle = (() => {
             if (disabled || !(overrides && overrides.textColor)) {
                 return textStyleMap[shape][theme][type];
+            } else {
```

Comment:
Having a return in the if statement stops un-needed nesting, and conforms to http://eslint.org/docs/rules/consistent-return (which I used, but not actually sure it's enabled in the project.